pkg/backend/httpstate/client/client.go

```diff
@@ -532,14 +532,19 @@ func (pc *Client) ImportStackDeployment(ctx context.Context, stack StackIdentifi
 	}, nil
 }
 
+type CreateUpdateDetails struct {
+	Messages         []apitype.Message
+	RequiredPolicies []apitype.RequiredPolicy
+}
+
 // CreateUpdate creates a new update for the indicated stack with the given kind and assorted options. If the update
 // requires that the Pulumi program is uploaded, the provided getContents callback will be invoked to fetch the
 // contents of the Pulumi program.
 func (pc *Client) CreateUpdate(
 	ctx context.Context, kind apitype.UpdateKind, stack StackIdentifier, proj *workspace.Project,
 	cfg config.Map, m apitype.UpdateMetadata, opts engine.UpdateOptions,
 	dryRun bool,
-) (UpdateIdentifier, []apitype.RequiredPolicy, error) {
+) (UpdateIdentifier, CreateUpdateDetails, error) {
```

Comment:
Is it ok to change this API? It's publicly exported.

Reply:
If relevant, from the doc.go for this package:

```go
// Package client implements a client for the Pulumi Service HTTP/REST API.
// Important note: This client is not versioned, and not intended for external use at this time.
package client
```